distribution/xfer/download.go

```diff
@@ -62,6 +63,9 @@ type DownloadDescriptor interface {
 	// if it is unknown (for example, if it has not been downloaded
 	// before).
 	DiffID() (layer.DiffID, error)
+	// ForeignSource returns nil or the foreign layer's information and
+	// source.
+	ForeignSource() *distribution.Descriptor
```

Comment:
@jstarks After discussing this with @aaronlehmann, we thought it would be okay as long as the ondisk formats would allow us to port this to other platforms. I think the only change that needs to be made is to call name the json file `descriptor.json` instead of `foreign-source.json`.